playwright/e2e/create-colony.spec.ts

```diff
@@ -0,0 +1,555 @@
+      // continue button is disabled initially
+      await expect(continueButton).toBeDisabled();
+
+      // Fill in colony name and custom URL
+      await page.getByLabel(/colony Name/i).fill(colonyName);
+      await page.getByLabel(/custom colony URL/i).fill(generateRandomString());
+      await page.getByLabel(/custom colony URL/i).blur();
+
+      // Check if URL is available
+      await expect(page.getByText(/URL available/i)).toBeVisible();
+
+      // Click continue and verify navigation to the next step
+      await continueButton.click();
+      await expect(
+        page.getByRole('heading', { name: /creating a new native token/i }),
+      ).toBeVisible();
+    });
+
+    test('Should show an error message when the Colony URL is already taken', async ({
+      page,
+    }) => {
+      // Mock the response for the 'CheckColonyNameExists' query
+      page.route('**/graphql', (route, request) => {
+        const postData = request.postDataJSON();
+
+        if (postData.operationName === 'CheckColonyNameExists') {
+          route.fulfill({
+            status: 200,
+            contentType: 'application/json',
+            body: JSON.stringify({
+              data: {
+                getColonyByName: {
+                  items: ['someColonyId'],
+                  __typename: 'ModelColonyConnection',
+                },
+              },
+            }),
+          });
+        } else {
+          route.continue();
+        }
+      });
+
+      const colonyUrlField = page.getByLabel(/custom colony URL/i);
+
+      await colonyUrlField.fill('takenurl');
+
+      await page.getByLabel(/colony name/i).focus();
+
+      await expect(
+        page.getByText(/This colony URL is already taken/i),
+      ).toBeVisible();
+    });
+
+    test.describe('Colony Name and URL validation', () => {
+      const validData = [
+        { field: 'colony name', values: validColonyNameFieldValues },
+        { field: 'custom colony URL', values: validColonyURLFieldValues },
+      ];
+
+      const invalidData = [
+        { field: 'colony name', values: invalidColonyNameFieldValues },
+        { field: 'custom colony URL', values: invalidColonyURLFieldValues },
+      ];
+
+      for (const { field, values } of validData) {
+        test(`Should accept valid ${field}`, async ({ page }) => {
+          const inputField = page.getByLabel(new RegExp(field, 'i'));
+
+          for (const value of values) {
+            await inputField.fill(value);
+            await inputField.blur();
+
+            await expect(page.getByTestId('form-error')).toBeHidden();
+          }
+        });
+      }
+
+      for (const { field, values } of invalidData) {
+        test(`Should reject invalid ${field}`, async ({ page }) => {
+          const inputField = page.getByLabel(new RegExp(field, 'i'));
+
+          for (const value of values) {
+            await inputField.fill(value);
+
+            await expect(page.getByTestId('form-error')).toBeVisible();
```

Comment:
**EDIT:** And now it passes 8 out of 10 times. It might just be my machine or something! The new thing that's flaking out is this test:

<img width="1800" alt="Screenshot 2024-10-30 at 03 21 58" src="https://github.com/user-attachments/assets/f695641a-eacd-4e55-99ac-8778d19365d6">

This might just work properly when run on CI. A good test to monitor but I have no objections on getting it merged!

Reply:
**EDIT:**

@olgapod Both tests have passed 5 out of 5 times on my machine without any code changes whatsoever 🚀